src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ProducerConsumerStream.cs

```diff
@@ -20,7 +20,6 @@ internal class ProducerConsumerStream : Stream
         public ProducerConsumerStream()
         {
             _buffer = new TaskCompletionSource<WriteBufferWrapper>();
-            _dataAvail = new TaskCompletionSource<int>();
```

Comment:
I'm a little concerned this field is no longer initialized here.  It is used in Flush without a null check, and because Flush is public, its possible it could be called before ReadAsync.  Unless there's a good reason, it would be better to guarantee initialization here or to null check before using it in Flush.